lab_1/main.py

```diff
@@ -3,81 +3,148 @@
 Language detection
 """
 
+import json
 
-def tokenize(text: str) -> list or None:
+
+def tokenize(text: str):
     """
     Splits a text into tokens, converts the tokens into lowercase,
     removes punctuation and other symbols from words
     :param text: a text
     :return: a list of lower-cased tokens without punctuation
-    """
-    pass
+        """
+    if isinstance(text, str):
+        text = text.lower()
+        string_with_tokens = ""
+        for i in text:
+            if i.isalpha() or i.isspace():
+                string_with_tokens += i
+        tokens = string_with_tokens.split()
+        return tokens
+    return None
 
 
-def remove_stop_words(tokens: list, stop_words: list) -> list or None:
+def remove_stop_words(tokens: list, stop_words: list):
     """
     Removes stop words
     :param tokens: a list of tokens
     :param stop_words: a list of stop words
     :return: a list of tokens without stop words
     """
-    pass
+    try:
+        for i in tokens:
+            if not isinstance(i, str):
+                return None
+            tokens = [i for i in tokens if i not in stop_words]
+            return tokens
+    except TypeError:
+        return None
 
 
-def calculate_frequencies(tokens: list) -> dict or None:
+def calculate_frequencies(tokens: list):
     """
     Calculates frequencies of given tokens
     :param tokens: a list of tokens
     :return: a dictionary with frequencies
     """
-    pass
+    if isinstance(tokens, list):
+        for i in tokens:
+            if isinstance(i, str):
+                freq_dict = {i: tokens.count(i) for i in tokens}
+                return freq_dict
+    return None
 
 
-def get_top_n_words(freq_dict: dict, top_n: int) -> list or None:
+def get_top_n_words(freq_dict: dict, top_n: int):
     """
     Returns the most common words
     :param freq_dict: a dictionary with frequencies
     :param top_n: a number of the most common words
     :return: a list of the most common words
     """
-    pass
-
-
-def create_language_profile(language: str, text: str, stop_words: list) -> dict or None:
+    if isinstance(freq_dict, dict) and isinstance(top_n, int):
+        if freq_dict != {} and top_n > 0:
+            new_dict = sorted(freq_dict.items(), key=lambda x: x[1], reverse=True)
+            count = 1
+            top_list = []
+            first, snd = zip(*new_dict)
+            del snd
+            for i in first:
+                if count <= top_n:
+                    top_list.append(i)
+                    count += 1
+                if count > top_n:
+                    break
```

Comment:
don't get this loop
maybe you can sort keys, not items, here? `new_dict = sorted(freq_dict.items(), key=lambda x: x[1], reverse=True)`